src/bundles/peer-locations.js

```diff
@@ -300,44 +148,65 @@ class PeerLocationResolver {
       ...opts.cache
     })
 
+    this.peersToAddrs = {}
+    this.failedAddrs = []
```

Comment:
`failedAddrs` and `peersToAddrs` are both unbounded caches... if we need them consider using an LRU cache with a max size.

Reply:
I'm just storing them to avoid calling the same lookup over and over despite of its failure. I've changed the max size to 500, where we delete the first 100 when it gets full. I think it's not worth it to implement a full LRU cache, but it follows the principle.